src/Button/index.js

```diff
@@ -0,0 +1,271 @@
+   const padding = props.padding && props.padding[breakpoint]
+   const size = props.size && props.size[breakpoint]
+
+   const height = buttonSizeDefaults[size].height
+   const width = props.width[breakpoint]
+
+   const textColor = props.textColor && props.textColor[breakpoint]
+   const textSize = props.textSize && props.textSize[breakpoint]
+   const textTransform = props.textTransform && props.textTransform[breakpoint]
+   const textWeight = props.textWeight && props.textWeight[breakpoint]
+
+   const primaryButtonColor = color || OIOContext.highlightColor
+   const defaultButtonPadding = width === 'auto' ? `0px ${height}` : '0px'
+
+   const buttonStyle = {
+      borderRadius,
+      fontFamily,
+      height,
+      textColor,
+      textTransform,
+      textWeight,
+      width,
+      backgroundColor: primaryButtonColor,
+      border: 'none',
+      textSize: textSize || buttonSizeDefaults[size].textSize,
+      hoverBackgroundColor: tinycolor(primaryButtonColor).lighten(7).toString(),
+      hoverBorder: '',
+      minWidth: height,
+      padding: padding ? `0px ${padding}` : defaultButtonPadding
+   }
+
+   if (rounded) {
+      buttonStyle.borderRadius = `${parseInt(height, 10) / 2}px`
+   }
+
+   if (outline) {
+      buttonStyle.border = `2px solid ${tinycolor(primaryButtonColor).setAlpha(0.5).toString()}`
+      buttonStyle.backgroundColor = 'transparent'
+      buttonStyle.hoverBackgroundColor = 'transparent'
+      buttonStyle.hoverBorder = `2px solid ${tinycolor(primaryButtonColor).setAlpha(0.8).toString()}`
+      buttonStyle.textColor = primaryButtonColor
+   }
+
+   return buttonStyle
+})
+
+// ============================================================================
+// Component
+// ============================================================================
+
+export default class Button extends React.Component {
+   static propTypes = {
+      borderRadius: OIOResponsiveObjectPropType,
+      children: PropTypes.node,
+      className: PropTypes.string,
+      color: OIOResponsiveObjectPropType,
+      fontFamily: OIOResponsiveObjectPropType,
+      mode: PropTypes.oneOf(['disabled', 'normal', 'pulsing', 'loading']),
+      name: PropTypes.node,
+      onClick: PropTypes.func,
+      outline: PropTypes.bool,
+      padding: OIOResponsiveObjectPropType,
+      rounded: PropTypes.bool,
+      size: OIOResponsiveObjectPropType,
+      style: PropTypes.object,
+      tagName: PropTypes.oneOf(['button', 'div', 'span']),
+      textColor: OIOResponsiveObjectPropType,
+      textSize: OIOResponsiveObjectPropType,
+      textTransform: OIOResponsiveObjectPropType,
+      textWeight: OIOResponsiveObjectPropType,
+      type: PropTypes.oneOf(['button', 'clear', 'submit']),
+      width: OIOResponsiveObjectPropType
+   }
+
+   static defaultProps = {
+      borderRadius: r`4px`,
+      children: undefined,
+      mode: 'normal',
+      onClick: () => {},
```

Comment:
Adds notable overhead given the number of buttons that may be instantiated. Should instead use `if` statements to check if `onClick` is defined.